### DISCUSSION TITLE
GH Actions - Timeout Approval for Workflow?

### DISCUSSION THREAD

**Discussion by ndom91** [community/community#29000] [Actions]
Hi everybody!

I noticed our GH Actions which weren't approved seem to hang around for the default [30 day timeout](https://docs.github.com/en/actions/managing-workflow-runs/reviewing-deployments#about-required-reviews-in-workflows). I'd like to kill these much earlier (~12hrs), but they're not idling in a workflow or job, they're waiting on approval from the environment protection rules.

I haven't found a way to set a `timeout-minutes` option for that specifically. Is it not possible?

If not, can you recommend anything else?

**Comment by thuongtruong1009** [comment]
Hi @ndom91 
You can learn control time to cron at [Crontab Guru](https://docs.getnacelle.com/deployment/scheduled-builds-github-actions.html#configure-the-schedule)

**Reply by ndom91** [reply]
I'm familiar with setting a cron-style trigger for GH Actions, but this is not what I want to achieve. The goal is to timeout "Waiting on Approval" jobs which have already been submitted, but not approved due to there being newer PRs, etc.

**Reply by ndom91** [reply]
Or are you suggesting running a separate GH Action on a schedule, like every 6 hrs and in it using the GH CLI or rest client to cancel those running GH Actions waiting on approval? Not such a bad idea actually haha, but more effort than its worth tbh :joy:

**Comment by marcin-sg** [comment]
Same problem here and also there #5673.

**Comment by oliversalzburg** [comment]
Setting a `concurrency` group with `cancel-in-progress` helps in some scenarios.

**Comment by benjiwolff** [comment]
We are also missing an option to set a default timeout period.

**Comment by Silva4g** [comment]
I have an similar case, I'm adding a concurrency and an reviewer, every time it's add a new pipeline it's stopped because one wasn't approve or deny, so I would like to add an time-out in the approval to resolve it

**Comment by bparthas21** [comment]
In the Setting -> Environments section, I see an option to set a timeout for approval.
![image](https://github.com/community/community/assets/88166247/38a54818-d961-49c5-a02d-6c80f439eb2b)

**Reply by xuxu314** [reply]
no, its not the same thing. timer is set for deployments to wait a certain period of time, then trigger deployment workflow

**Reply by bparthas21** [reply]
My understanding of OPs question is
- Workflow triggers and waits for approval.
- OP wants the workflow to proceed without approval after X amount of time (aka timeout).

If that's correct, isn't the checkbox "Wait timer" meant for that purpose? (that's what the description seems to indicate).

**Reply by degould-mm** [reply]
The OP wants to kill the deployment, which is awaiting approval, before the 30 day timeout. The wait timer is to delay the deployment for x minutes. Two different things.

**Comment by humzam** [comment]
We are unhappy with the current solution as well. The default timeout of 30 days is incredibly long. Would really like the ability to configure deployment approval timeouts.

**Reply by donald-berning** [reply]
We have the opposite problem most are noting. Our build/deploy process can span over 30 days until finally in production. A configurable solution at the workflow action run level (longer for production candidates, shorter for development only) would be ideal.

**Comment by oliora** [comment]
+1

**Comment by defyjoy** [comment]
Honestly , the jobs should be cancelled and not failed . Having failed job shows all the actions tab as failed ( red icon ) and it is not very intuitive to know if ACTUALLY the job failed due to error pipeline error or it's just cancelled. 

Having a timeout to cancel the job instead so that the pipeline shows as cancelled ( grey icon ) and not error ( Red icon )

**Comment by mikebollandajw** [comment]
timeout-minutes doesnt seem to cancel my approval, any idea?

**Comment by strachg** [comment]
I'm disappointed but not surprised at this situation. I've just started using workflow approvals and noticed the issue. Get on it Github 

This thread provides some HACKS to make github do what it should, by creating that other workflow to keep an eye on workflows that have been waiting too long. Sorry, no way to extend past 30days from what I can see.

https://github.com/orgs/community/discussions/50926#discussioncomment-6161198
